### DISCUSSION TITLE
TypeError: this.firebaseApp.functions is not a function when upgrading to firebase 9; angularFire 7

### DISCUSSION THREAD

**Discussion by LanderBeeuwsaert** [angular/angularfire#2950] [Q&A] [answered]
Hey all, while trying to get the upgrade working, seems that the way I'm calling a function doesn't work anymore.
As I still get autocomplete typings for this (this.firebaseApp.functions()) I would still expect this to work? (see below)

### Version info

**Angular:**
12.2.0

**Firebase:**
9.0.1

**AngularFire:**
7.0.4

**Other (e.g. Ionic/Cordova, Node, browser, operating system):**
windows 10, chrome, node 16, npm 7

### How to reproduce these conditions

The code I have (I've shortened it a bit, but the important things are there):

import { FirebaseApp } from '@angular/fire/compat';

@Injectable()
export class SomeService {
  constructor(
    private firebaseApp: FirebaseApp,
  ) {
    await this.firebaseApp.functions().httpsCallable('addOrganisation')(addOrganisationParams);

This gives the error:
TypeError: this.firebaseApp.functions is not a function

[a comment by an automated account is not shown]

**Comment by jamesdaniels** [comment, marked as answer]
You should inject functions, [`AngularFireFunctions` in your constructor ](https://github.com/angular/angularfire/blob/master/samples/compat/src/app/functions/functions.component.ts#L20)and make sure your [importing `AngularFireFunctionsModule` in your AppModule](https://github.com/angular/angularfire/blob/master/samples/compat/src/app/app.module.ts#L68) for best experience. It's likely something is getting tree-shaken out, or your "side-effect" import (`import 'firebase/functions'`) was not converted to compat, injecting `AngularFireFunctions` means you don't have to worry about any of that.
